pylabrobot/resources/rotation.py

```diff
@@ -1,36 +1,47 @@
 import math
-
+from pylabrobot.resources.coordinate import Coordinate
 from pylabrobot.utils.linalg import matrix_multiply_3x3
 
-
 class Rotation:
   """ Represents a 3D rotation. """
 
   def __init__(self, x: float = 0, y: float = 0, z: float = 0):
-    self.x = x  # around x-axis, roll
-    self.y = y  # around y-axis, pitch
-    self.z = z  # around z-axis, yaw
+    self.x = x 
+    self.y = y
+    self.z = z
 
-  def get_rotation_matrix(self):
-    # Create rotation matrices for each axis
-    Rz = ([
+  def get_rotation_and_translation(self, origin: Coordinate = Coordinate.zero()):
+    # rotation matrices for each axis
+    Rz = [
       [math.cos(math.radians(self.z)), -math.sin(math.radians(self.z)), 0],
       [math.sin(math.radians(self.z)), math.cos(math.radians(self.z)), 0],
       [0, 0, 1]
-    ])
-    Ry = ([
+    ]
+    Ry = [
       [math.cos(math.radians(self.y)), 0, math.sin(math.radians(self.y))],
       [0, 1, 0],
       [-math.sin(math.radians(self.y)), 0, math.cos(math.radians(self.y))]
-    ])
-    Rx = ([
+    ]
+    Rx = [
       [1, 0, 0],
       [0, math.cos(math.radians(self.x)), -math.sin(math.radians(self.x))],
       [0, math.sin(math.radians(self.x)), math.cos(math.radians(self.x))]
-    ])
+    ]
     # Combine rotations: The order of multiplication matters and defines the behavior significantly.
     # This is a common order: Rz * Ry * Rx
-    return matrix_multiply_3x3(matrix_multiply_3x3(Rz, Ry), Rx)
+    rotation_matrix = matrix_multiply_3x3(matrix_multiply_3x3(Rz, Ry), Rx)
+
+    origin_vector = origin.vector()
+    rotated_origin = [
+      sum(rotation_matrix[i][j] * origin_vector[j] for j in range(3)) for i in range(3)
+    ]
+    translation = [
+      rotated_origin[0] - origin.x,
+      rotated_origin[1] - origin.y,
+      rotated_origin[2] - origin.z
+    ]
+
+    return rotation_matrix, translation
```

Comment:
we need to return the rotation because otherwise it would break get_absolute_rotation and other functions elsewhere

Reply:
hmm idk how to do this without a 4x4 matrix or a second translation matrix in addition to the rotation matrix

4x4 would just add the translation as a column, but would then have to be implemented in get_absolute_rotation and others?
```
rotation_matrix_4x4 = [
            [rotation_matrix_3x3[0][0], rotation_matrix_3x3[0][1], rotation_matrix_3x3[0][2], translation.x],
            [rotation_matrix_3x3[1][0], rotation_matrix_3x3[1][1], rotation_matrix_3x3[1][2], translation.y],
            [rotation_matrix_3x3[2][0], rotation_matrix_3x3[2][1], rotation_matrix_3x3[2][2], translation.z],
            [0, 0, 0, 1]  # Last row for homogeneous transformation
        ]
```